docs/form-app-script.js

```diff
@@ -0,0 +1,120 @@
+}
+
+function getOrCreateSponsor(attributes) {
+  if (!attributes['Existing sponsor'].match(/\Other/)) {
+    return {id: attributes['Existing sponsor'].split(':')[1].trim()}
+  } else {
+    var newSponsor = {
+      sponsor: {
+        name: attributes['New sponsor name'],
+        billing_info: attributes['New sponsor billing information'],
+        notes: attributes['Notes about new sponsor']
+      }
+    }
+    var sponsorResponse = JSON.parse(postRequest(SPONSOR_URL, newSponsor).getContentText());
+    return {id: sponsorResponse.id};
+  }
+}
+
+function getOrCreatePodcast(attributes) {
+  if (!attributes['Existing podcast'].match(/\Other/)) {
+    return {id: attributes['Existing podcast'].split(':')[1].trim()}
+  } else {
+    var newPodcast = {
+      show: {
+        name: attributes['New podcast name'],
+        network: attributes['New podcast network'],
+        structure: attributes['New podcast show structure'],
+        recording_day: attributes['New podcast recording day of week (if known)'],
+        rate: attributes['New podcast rate'],
+        notes: attributes['Notes about new podcast']
+      }
+    }
+    var podcastResponse = JSON.parse(postRequest(PODCAST_URL, newPodcast).getContentText());
+    return {id: podcastResponse.id};
+  }
+}
+
+function createCampaign(attributes) {
+  var newCampaign = {
+    campaign: {
+      start_date: attributes['Start date'],
+      end_date: attributes['End date'],
+      copy: attributes['Copy'],
+      zone: attributes['Zone(s)'].join(),
+      sponsor_id: attributes['sponsor_id'],
+      show_id: attributes['show_id']
+    }
+  }
+  postRequest(CAMPAIGN_URL, newCampaign);
+}
+
+function postRequest(url, payload){
+  var request = {
+    method: 'post',
+    contentType: 'application/json',
+    payload:  JSON.stringify(payload)
+  };
+  return UrlFetchApp.fetch(url, request);
+}
+
+// code below handles the setting of the "existing podcast" and "existing sponsor" choices to match what's in Jingle
+// TODO should change these to pull from Adzerk sites + advertizers (networks too)
+
+
+function setChoices() {
+  var questions = FormApp.getActiveForm().getItems();
+  questions.forEach(function (question) {
+    if (question.getTitle() == "Existing sponsor" && question.getType() == FormApp.ItemType.MULTIPLE_CHOICE) {
+      setAnswerChoices(question.asMultipleChoiceItem(), SPONSOR_URL);
+    }
+    if (question.getTitle() === "Existing podcast" && question.getType() == FormApp.ItemType.MULTIPLE_CHOICE) {
+      setAnswerChoices(question.asMultipleChoiceItem(), PODCAST_URL);
+    }
+  });
+}
+
+function setAnswerChoices(question, url){
+  var titlesAndIds = fetchJingleResources(url).map(function (r) { return r.name + ": " + r.id });
+  question.setChoiceValues(titlesAndIds.concat('Other (enter new info below)'));
+}
+
+function fetchJingleResources(url){
+  return JSON.parse(UrlFetchApp.fetch(url).getContentText())["_embedded"]["prx:items"].map(function (el) {
```

Comment:
So I'm guessing eventually you'll have to provide some sort of auth to this endpoint?  Or is the plan to leave parts of Jingle as public-read?

Reply:
nope, down the line we'll add authentication of some kind to protect this endpoint. for the purposes of this form, currently only people with a PRX email address can access it.

Reply:
(I commented out the auth pipeline in the jingle router for purposes of getting this up and running)

Reply:
Gotcha.  We could do something like generate an auth-token with an incredibly long TTL, and plug that into the script when pasting it into the script editor.  But can handle that later on!